.vscode/tasks.json

```
@@ -1,58 +1,43 @@
+
 {
   "tasks": [
     {
+      "label": "WaitForDebugpy",
       "type": "shell",
-      "label": "conda: activate environment",
-      "command": "activate freecad",
+      "command": "python ${workspaceFolder}/.vscode/WaitForPort.py",
+      "group": "none",
       "problemMatcher": [],
-      "detail": "Activate conda environment",
-      "group": "build"
+      "presentation": {
+        "reveal": "never", //silently fail and don't launch the debugger
+        "panel": "dedicated",
+        "close": true,
+        "revealProblems": "never"
+      },
+      "hide": true
     },
     {
-      "type": "cmake",
+      "type": "shell",
       "label": "CMake: configure",
-      "command": "configure",
+      "windows": { "command": "if ('${command:cmake.activeConfigurePresetName}'.ToLower().Contains('conda')) { activate freecad };               cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
+      "linux":   { "command": "shopt -s nocasematch; [[ ${command:cmake.activeConfigurePresetName} =~ \"conda\" ]] && source activate freecad; cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
+      "osx":     { "command": "shopt -s nocasematch; [[ ${command:cmake.activeConfigurePresetName} =~ \"conda\" ]] && source activate freecad; cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
```

```suggestion
      "windows": { "command": "if ('${command:cmake.activeConfigurePresetName}'.ToLower().Contains('conda')) { conda activate freecad };               cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
      "linux":   { "command": "shopt -s nocasematch; [[ ${command:cmake.activeConfigurePresetName} =~ \"conda\" ]] && conda activate freecad; cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
      "osx":     { "command": "shopt -s nocasematch; [[ ${command:cmake.activeConfigurePresetName} =~ \"conda\" ]] && conda activate freecad; cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
```

this is safer IMO, in my conda setup the base environment is not automatically activated on launch (personal preference) so `activate` is not directly available. @oursland is there a difference when using `source activate` instead of `conda activate`?

---

Here's what I got when I switched to `conda activate`:

     *  Executing task: shopt -s nocasematch; [[ conda-release =~ "conda" ]] && conda activate freecad; cmake .../FreeCAD --preset conda-release 

    zsh:1: command not found: shopt

    CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
    To initialize your shell, run

        $ conda init <SHELL_NAME>

    Currently supported shells are:
      - bash
      - fish
      - tcsh
      - xonsh
      - zsh
      - powershell

    See 'conda init --help' for more information and options.

    IMPORTANT: You may need to close and restart your shell after running 'conda init'.

Using `conda init bash` and `conda init zsh`, to ensure both `bash` and `zsh` are initialized did not resolve the issue.

I suspect VS Code is executing the shell without inheriting the environment and without processing profile and rc files.

---

> zsh:1: command not found: shopt

ah, zsh has `setopt`. I need to find a better solution that works for all shells

---

See https://stackoverflow.com/questions/52748203/using-a-bash-profile-function-in-vs-code-tasks